api/src/main/java/com/codeforcommunity/rest/ApiRouter.java

```diff
@@ -467,24 +492,40 @@ private void handleCreateEvent(RoutingContext ctx) {
 
     try {
       name = body.getString("name");
+      subtitle = body.getString("subtitle");
+      description = body.getString("description");
+      imageUrl = body.getString("imageUrl");
       date = LocalDateTime.parse(body.getString("date"), formatter);
       open = body.getBoolean("open");
       eventCode = body.getString("code");
     } catch (Exception e) {
       e.printStackTrace();
       response.setStatusCode(400).end();
     }
-    boolean success = false;
-    if (name != null && date != null && open != null && eventCode != null)
-      success = processor.createEvent(name, date, open, eventCode);
+    //-----------------------DUPLICATE FOR OTHER TYPES------------------
 
-    if (success)
-      response.setStatusCode(201).end();
+    Optional<EventReturn> result = Optional.empty();
```

Comment:
Seems like a generic helper function would be able to get rid of a good amount of duplication here